optaplanner-core/src/main/java/org/optaplanner/core/impl/io/XmlIO.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.io;
+
+import java.io.Reader;
+import java.io.Writer;
+
+/**
+ * Generic XML read/write facility.
+ */
+public interface XmlIO<T> {
```

Comment:
Personally, I don't see a need for this. It's in `impl`, so we needn't worry about backwards compatibility or exposing public methods. You might've as well used the class directly.

Reply:
As in my answer to the previous comment, the main motivation is to avoid leaking JAXB and make it easily replaceable by any future XML marshalling technology, if needed.

Reply:
XML is dying, cool kids use JSON now, there won't be any other XML technology. :-)

Reply:
I agree the interface is not necessary for our purposes, will remove it.

But I disagree with the JSON, cool kids use YAML nowadays.